### PR TITLE
LGA-1776 update django restframework to 3.3.3

### PR DESCRIPTION
**Pull request by MishNajam**
## What does this pull request do?

- Pagination has changed for 3.3.3. Rather than use properties of the serializer viewset, these now have to be declared via a pagination class. You cannot use the base pagination class, you have to override and declare your own. The base pagination class has key properties set as none. This means that even if you declared pagination in the viewset by declaring `pagination_class=PageNumberPagination` there would be no pagination.  

- ExportListView had a page_size of 1000 set but this did not seem to be doing anything - when we added pagination in for that viewset, the view did not work as expected. Hence `page_size` was removed and no pagination included

- A manual test to check that more than 1000 rows did not change behaviour between master and this branch was performed.

- ArticleViewSet in checker had settings for pagination that mirrored those in BaseArticleViewSet so these were removed as not adding anything above the base class.

## Any other changes that would benefit highlighting?

- Fixed language versions in the pre-commit.yaml for all of the hooks - this helps when developers may have different versions of pre-commit installed.
- We used new database diff tool to check for any changes not picked up by the tests and this looked fine. Have agreed to create an annotation/md file for the end to end tests repository to explain why there will currently be some differences. This annotation can then be used to update the diff tool so that it can automatically accept these differences - a JIRA card will be created.

## Checklist

- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: cla_backend/apps/reports/api.py [hunk 1]
```diff
@@ -14,7 +13,6 @@ class ExportListView(generics.ListAPIView, generics.DestroyAPIView):
     model = Export
     permission_classes = (IsAdminUser,)
     authentication_classes = (SessionAuthentication,)
-    page_size = 1000
```
Comment (said-moj) [inline cla_backend/apps/reports/api.py:17]: Why wasn't `pagination_class` added to this viewset?

File: cla_backend/apps/checker/views.py [hunk 2]
```diff
@@ -44,8 +44,6 @@ class Meta(ArticleCategoryFilter.Meta):
 
 
 class ArticleViewSet(PublicAPIViewSetMixin, BaseArticleViewSet):
-    paginate_by_param = "page_size"
```
Comment (said-moj) [inline cla_backend/apps/checker/views.py:47]: Why wasn't `pagination_class` added to this viewset?

File: cla_backend/apps/reports/api.py (hunk 1 above)
Comment (jogilder) [reply cla_backend/apps/reports/api.py:17]: We don't think that this one was being used in master - we added in pagination and then took it out again as adding in pagination actually resulted in a difference to master. We did a one off automated test adding in over 1000 rows to the table of exports and neither master or the new branch seemed to do any pagination.

File: cla_backend/apps/checker/views.py (hunk 2 above)
Comment (jogilder) [reply cla_backend/apps/checker/views.py:47]: Not sure, I will follow up on this one

Comment (jogilder) [reply cla_backend/apps/checker/views.py:47]: Checked and this was not doing anything - BaseArticleViewSet already declares the same pagination details

**Review by Pheonnexx — APPROVED** [review]
awesome work 🚀

**Review by said-moj — APPROVED** [review]
👍 LGTM

**Pull request merged**